lib/FS_unix.ml

```diff
@@ -132,6 +137,13 @@ let stat {base} path0 =
   with exn ->
     return (`Error (`No_directory_entry (base, path0)))
 
+let connect id =
+  try_lwt
+    match Sys.is_directory id with
+    | true -> return (`Ok {base = id})
+    | false -> return (`Error (`Not_a_directory id))
+  with (Sys_error _) -> return (`Error (`No_directory_entry (id, "")))
```

Comment:
Would be nice to use something like `check_type` from `create_directory` here and then catch the `Unix_error` and use the `create_directory` error map so things like perms errors are handled correctly (right now, will be effectively `ENOENT`).